docs-chef-io/content/server/upgrade_server_ha_v2.md

```diff
@@ -161,15 +161,203 @@ Upgrading from Chef Backend 1.x to Chef Backend 2.x requires full cluster downti
+| 2.3.16 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.2.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.1.0 | 3.0 | `"created_string" : "5.6.16"` | Automatic reindex |
+| 2.0.30 | 2.3.16 | `"created_string" : "5.4.1"` | Delete and reindex |
+| 1.4.6 | 2.3.16 | `"created_string" : "2.3.1"` | Delete and reindex |
+| 1.4.6 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.3.2 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.2.5 | 2.3.16 | `"created_string" :` Any  | Delete and reindex |
+| 1.1.2 | 2.3.16 | `"created_string" :` Any | Delete and reindex |
+| 1.0.2 | 2.3.16 | `"created_string" :` Any | Delete and reindex |
+
+### Upgrade instructions for servers with Elasticsearch 1.7-5.4.1
+
+This requires downtime and the indexes are deleted and reindexed.
+
+#### Front-end
+
+1. Stop all front-end nodes. This is to make sure that we do not make any calls from the front-end to the back-end while we delete and reindex. 
+
+    1. On Chef Infra Server 14.6.32 or later, put the server nodes into maintenance mode, which temporarily disables the API temporarily. if
+
+      ```bash
+      sudo chef-server-ctl maintenance on
+      ```
+
+    1. On Chef Infra Server 14.6.31 or earlier, stop all server nodes with the command:
+
+      ```bash
+      sudo chef-server-ctl stop
+      ```
+
+#### Back-end
+
+1. Identify the **leader** node:
+
+    Find **leader** node in the **Role** column in the output of the `cluster-status` subcommand. In this example, `backend-1` is the **leader** node, as indicated by its role in the **Role** column.
+
+    ```ruby
+    chef-backend-ctl cluster-status
+    ```
+
+    ```output
+    Name       IP              GUID                              Role      PG        ES
+    backend-1  192.168.33.215  dc0c6ea77a751f94037cd950e8451fa3  leader    leader    not_master
+    backend-2  192.168.33.216  008782c59d3628b6bb7f43556ac0c66c  follower  follower  not_master
+    backend-3  192.168.33.217  1af654172b1830927a571d9a5ba7965b  follower  follower  master
+    ```
+
+2. Install the new Chef Backend package on each of the server nodes in the cluster:
+
+    RHEL and Centos:
+
+    ```bash
+    yum install PATH_TO_FILE.rpm
+    ```
+
+    Debian and Ubuntu:
+
+    ```bash
+    dpkg -i PATH_TO_FILE.deb
+    ```
+
+3. Next, run the upgrade subcommand on each of the follower nodes:
+
+    ```bash
+    chef-backend-ctl upgrade
+    ```
+
+5. After finishing the upgrades on the follower nodes, **then** upgrade the leader node, using  the `--failover` option:
+
+    ```bash
+    chef-backend-ctl upgrade --failover
+    ```
+
+6. Ensure that Chef Backend creates new indexes using the latest installed Elasticsearch version. Execute each of these three commands on every Chef Backend server node:
+
+   ```bash
+   sudo chef-backend-ctl stop elasticsearch
+   sudo rm -fr /var/opt/chef-backend/elasticsearch/data/*
+   sudo chef-backend-ctl start elasticsearch
+   ```
```

Comment:
And finally, turn on failover again https://docs.chef.io/server/upgrade_server_ha_v2/#step-4-re-enable-failover